source/Nevermore/Advanced/QueryBuilders/JoinSourceQueryBuilder.cs

```diff
@@ -38,7 +38,7 @@ protected override ISelectBuilder CreateSelectBuilder()
             return new JoinSelectBuilder(joinedSource);
         }
 
-        public override IJoinSourceQueryBuilder<TRecord> Join(IAliasedSelectSource source, JoinType joinType, CommandParameterValues parameterValues, Parameters parameters, ParameterDefaults parameterDefaults)
+        public override IJoinSourceQueryBuilder<TRecord> Join(IAliasedSelectSource source, JoinType joinType, CommandParameterValues parameterValues, Parameters parameters, ParameterDefaults parameterDefaults, string queryAlias = null)
```

Comment:
Not sure of the use of the `Join` method here, the `JoinSourceQueryBuilder` purpose seems to be towards setting up the `ON` statement, so the function and the additional `queryAlias` here serve no purpose, unless I am missing something.

Reply:
Yeah, this one looks a bit awkward, like the abstraction isn't quite right.